compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -42,7 +42,11 @@ impl<'tcx> Const<'tcx> {
         let hir_id = tcx.hir().local_def_id_to_hir_id(def.did);
 
         let body_id = match tcx.hir().get(hir_id) {
-            hir::Node::AnonConst(ac) => ac.body,
+            hir::Node::AnonConst(ac)
+            | hir::Node::GenericParam(hir::GenericParam {
+                kind: hir::GenericParamKind::Const { ty: _, default: Some(ac) },
+                ..
```

Comment:
I wonder if some of the problems come from the fact that `const_param_default` can be invoked on either the default or the parameter, and will give you the default's value either time.

Reply:
yeah, added this so that i am able to quickly use this method for the default.

Now that we have a separate query for that it's probably good to move this change into that query to get the inner `DefId`

Reply:
Should I just remove this match branch then? And move the rest of this fn into the query?

Reply:
My idea is to remove revert `from_anon_const` to how it was before this pr and change `const_param_default` to call `from_anon_const` with the `DefId` of the default, so something like
```rust
fn const_param_default(tcx: TyCtxt, const_param_def_id: DefId) -> ... {
     let default_def_id = match tcx.hir().get(...) { ... };
     ty::Const::from_anon_const(default_def_id)
}
```
was that what you've meant here?

Reply:
ah no, to be quite honest I was completely unsure what you were thinking, so just threw out something random, but this sounds good.